localstack/services/awslambda/api_utils.py

```diff
@@ -69,10 +72,14 @@
 SIGNING_PROFILE_VERSION_ARN_REGEX = re.compile(
     r"arn:(aws[a-zA-Z0-9-]*):([a-zA-Z0-9\-])+:([a-z]{2}(-gov)?-[a-z]+-\d{1})?:(\d{12})?:(.*)"
 )
+# Combined pattern for alias and version based on AWS error using "(|[a-zA-Z0-9$_-]+)"
+QUALIFIER_REGEX = re.compile(r"(^[a-zA-Z0-9$_-]+$)")
 # Pattern for a version qualifier
 VERSION_REGEX = re.compile(r"^[0-9]+$")
 # Pattern for an alias qualifier
-ALIAS_REGEX = re.compile(r"(?!^[0-9]+$)([a-zA-Z0-9-_]+)")
+# Rules: https://docs.aws.amazon.com/lambda/latest/dg/API_CreateAlias.html#SSS-CreateAlias-request-Name
+# The original regex from AWS misses ^ and $ in the second regex, which allowed for partial substring matches
+ALIAS_REGEX = re.compile(r"(?!^[0-9]+)(^[a-zA-Z0-9-_]+$)")
```

Comment:
Love the addition of the source here! Makes it much easier later on to trace back the decision for certain things

Reply:
Thank you 🙏